src/presetsController.ts

```diff
@@ -1129,12 +1540,25 @@ export class PresetsController {
     }
 
     // Note: in case anyone want to change this, presetType must match the corresponding key in presets.json files
-    async addPresetAddUpdate(newPreset: preset.ConfigurePreset, presetType: 'configurePresets' | 'buildPresets' | 'testPresets') {
+    async addPresetAddUpdate(newPreset: preset.ConfigurePreset | preset.BuildPreset | preset.TestPreset | preset.PackagePreset | preset.WorkflowPreset,
+        presetType: 'configurePresets' | 'buildPresets' | 'testPresets' | 'packagePresets' | 'workflowPresets') {
         const originalPresetsFile: preset.PresetsFile = preset.getOriginalPresetsFile(this.folderPath) || { version: 2 };
         if (!originalPresetsFile[presetType]) {
             originalPresetsFile[presetType] = [];
         }
-        originalPresetsFile[presetType]!.push(newPreset);
+
+        switch (presetType) {
+            case "configurePresets":
+            case "buildPresets":
+            case "testPresets":
+            case "packagePresets":
+                originalPresetsFile[presetType]!.push(newPreset);
```

Comment:
Why the difference in these two cases? But the other cases don't need differences?

Reply:
I had red squiggle because:

"Argument of type 'ConfigurePreset | BuildPreset | TestPreset | PackagePreset | WorkflowPreset' is not assignable to parameter of type 'WorkflowPreset'. Property 'steps' is missing in type 'ConfigurePreset' but required in type 'WorkflowPreset'."

If steps were not a required field I think the ".push" would work for all 5 preset types.